### PR TITLE
qa: cleanup MLflow

### PR DESCRIPTION
**Pull request by aarnphm**
Cleanup some logic, qa with local postgres, mysql setup of mlflow registry.
cleanup some types as well.

### CODE REVIEW

File: bentoml/_internal/frameworks/mlflow.py [hunk 1]
```diff
@@ -4,7 +4,6 @@
 import shutil
 import typing as t
 import logging
-import tempfile
```
Comment (parano) [inline bentoml/_internal/frameworks/mlflow.py:7]: this is still required, right?

Comment (aarnphm) [reply bentoml/_internal/frameworks/mlflow.py:7]: I imported this lazily

Comment (parano) [reply bentoml/_internal/frameworks/mlflow.py:7]: probably no need to do that for built-in libraries

File: bentoml/_internal/frameworks/mlflow.py [hunk 2]
```diff
@@ -205,19 +206,20 @@ def import_model(
...
-        # If the temp dir we created still exists now, we never needed it because
-        # the provided mlflow url must have provided enough path information.  Just
-        # delete it.  If it's not here, it means we needed it but it's been renamed
-        # by now so we don't need to remove it.
-        if os.path.isdir(download_dir):
+        finally:
+            if TYPE_CHECKING:
+                # We need to trick type checker here, since local_path
+                # will always be set
+                local_path = ""
```
Comment (ssheng) [inline bentoml/_internal/frameworks/mlflow.py:216]: I think we have gone too far to please the type checker. 🙂 Maybe put this before the `try`?

File: bentoml/_internal/frameworks/mlflow.py [hunk 3]
```diff
@@ -205,19 +206,20 @@ def import_model(
...
-
-        mlflow_model_path = bento_model.path_of(MLFLOW_MODEL_FOLDER)
-        # Rename model folder from original artifact name to fixed "mlflow_model"
-        shutil.move(local_path, mlflow_model_path)
-        # If the temp dir we created still exists now, we never needed it because
-        # the provided mlflow url must have provided enough path information.  Just
-        # delete it.  If it's not here, it means we needed it but it's been renamed
-        # by now so we don't need to remove it.
-        if os.path.isdir(download_dir):
+        finally:
```
Comment (ssheng) [inline bentoml/_internal/frameworks/mlflow.py:212]: 👍

File: bentoml/_internal/frameworks/mlflow.py (hunk 2 above)
Comment (aarnphm) [reply bentoml/_internal/frameworks/mlflow.py:216]: 😄

Comment (sauyon) [reply bentoml/_internal/frameworks/mlflow.py:216]: I think this is a true positive; it's possible if there are exceptions raised that `local_path` is not set.

**Review by ssheng — APPROVED** [review]

**Pull request merged**